### PR TITLE
fix: fix the order of pinned icons in profile

### LINKED ISSUE #8533: [BUG] The order of pinned icons doesn't change.
### Description

After ordering the links, the order of pinned links doesn't change; it remains the same.

Social shortcut icons should be ordered by the links order

Screenshot 2023-08-07 at 10 02 59

### Screenshots

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by anikdhabal**
## Fixes Issue
fixes #8533 

https://github.com/EddieHubCommunity/LinkFree/assets/81948346/16912b97-42fb-469b-8142-32fa92134e9e

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8687"></a>

### CODE REVIEW

File: components/user/UserProfile.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ function UserProfile({ BASE_URL, data }) {
         <div className="flex flex-col self-center gap-3">
           <h1 className="text-3xl font-bold">{data.name}</h1>
           <div className="flex md:w-full gap-2 mx-auto text-xl">
-            {data.socials?.map((social) => (
+            {data.links?.map((social) => (
```
Comment (eddiejaoude) [inline components/user/UserProfile.js:55]: Not all `links` are pinned as `socials`

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution but this will break the functionality

Please look at making changes in the API where the links are ordered the socials can be ordered also

**Comment by anikdhabal** [comment]
> Thank you for your contribution but this will break the functionality
> 
> Please look at making changes in the API where the links are ordered the socials can be ordered also

Understood,Thank you.

**Comment by anikdhabal** [comment]
> Thank you for your contribution but this will break the functionality
> 
> Please look at making changes in the API where the links are ordered the socials can be ordered also

Screenshot 2023-09-05 201808
I've updated the socials also, but it's not working. @eddiejaoude

File: components/user/UserProfile.js (hunk 1 above)
Comment (ckelwin) [reply components/user/UserProfile.js:55]: The socials are not sorted properly in `getUserApi`.

**Comment by eddiejaoude** [comment]
What is not working? Please push your changes

**Comment by anikdhabal** [comment]
> What is not working? Please push your changes

Ok

**Comment by anikdhabal** [comment]
@eddiejaoude pls check it once.

**Comment by eddiejaoude** [comment]
I think this needs to be done in the API when retrieving the data - as the data is already saved correctly

**Comment by SaraJaoude** [comment]
Due to lack of activity I will close this PR.

**Pull request closed without merging**